backend/api/Areas/Tools/Helpers/ImportProjectsHelper.cs

```diff
@@ -203,14 +300,100 @@ private Entity.Project Merge(Entity.Project project, Model.ImportProjectModel mo
+        /// Parse the 'noteType' value, or set it as 'General'.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <param name="noteType"></param>
+        /// <param name="note"></param>
+        private void AddNote(Entity.Project project, string noteType, string note)
+        {
+            if (!String.IsNullOrWhiteSpace(note))
+            {
+                if (!Enum.TryParse(noteType, true, out Entity.NoteTypes type))
+                {
+                    type = noteType switch
+                    {
+                        "Property" => Entity.NoteTypes.Public,
+                        "PID" => Entity.NoteTypes.Public,
+                        "LandLegalDescription" => Entity.NoteTypes.Public,
+                        "Surplus Dec / Readiness Checklist" => Entity.NoteTypes.Public,
+                        "TBL - CBA" => Entity.NoteTypes.Public,
+                        "Enhanced Referral Comments" => Entity.NoteTypes.Private,
+                        "Interest from Enhanced Referral" => Entity.NoteTypes.AgencyInterest,
+                        "Business Case Comments" => Entity.NoteTypes.Private,
+                        "Business Case Submitted" => Entity.NoteTypes.Private,
+                        "Clearance Letter Issued" => Entity.NoteTypes.Private,
+                        "Appraisal Date" => Entity.NoteTypes.Appraisal,
+                        "Assessed & Appraised Comments" => Entity.NoteTypes.Appraisal,
+                        "Weekly Review" => Entity.NoteTypes.Private,
+                        _ => Entity.NoteTypes.General
+                    };
+                    note = $"{noteType}{Environment.NewLine}{note}";
+                }
+
+                var projectNote = project.Notes.FirstOrDefault(n => n.NoteType == type);
+                if (projectNote == null)
+                {
+                    project.Notes.Add(new Entity.ProjectNote(project, type, note));
+                }
+                else
+                {
+                    if (!String.IsNullOrWhiteSpace(projectNote.Note))
+                        projectNote.Note += $"{Environment.NewLine}{Environment.NewLine}{note}";
+                    else
+                        projectNote.Note = note;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Find the property for the specified 'pid' in inventory and add it to the project.
+        /// Assumption is made that if the parcel is added, then also add all the buildings on the parcel.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <param name="pid"></param>
+        /// <returns></returns>
+        private Entity.Project AddProperty(Entity.Project project, string pid)
+        {
+            try
+            {
+                var pidValue = Int32.Parse(pid.Replace("-", ""));
+                var parcel = _adminService.Parcel.GetByPid(pidValue);
+                project.AddProperty(parcel).ForEach(p => {
```

Comment:
I'm finding the two functions both called AddProperty to be confusing.

Reply:
The `AddProperty(...)` extension function just does a handful of required steps for adding either `parcel` or `building` to a `project`.  In this case because we're dealing with unattached entities if we reference them it results in attempts to add duplicate values to the database.  Which either results in actually inserting duplicate records into the database, or an error because it already exists in the database.